pypdf/generic/_base.py

```diff
@@ -409,8 +412,8 @@ def clone(
     def myrepr(self) -> str:
         if self == 0:
             return "0.0"
-        nb = int(log10(abs(self)))
-        s = f"{self:.{max(1,16-nb)}f}".rstrip("0").rstrip(".")
+        nb = FLOAT_WRITE_PRECISION - int(log10(abs(self)))
+        s = f"{self:.{max(1,nb)}f}".rstrip("0").rstrip(".")
```

Comment:
```suggestion
        nb = int(log10(abs(self)))
        s = f"{self:.{max(1, FLOAT_WRITE_PRECISION - nb)}f}".rstrip("0").rstrip(".")
```

Reply:
I thought it would be easier to understand to get n as the number of digit to be display

Reply:
I think it was easier the other way around, but that's fine. It's certainly opinionated and I'm open to change there :-) I'll merge it as-is :-)